apollo-router/tests/integration/directives/policy.rs

```diff
@@ -0,0 +1,604 @@
+    //   * the supergraph service layer
+    //   * a request for a policy-gated field
+
+    let mock_coprocessor = MockServer::start().await;
+    let coprocessor_address = mock_coprocessor.uri();
+
+    Mock::given(method("POST"))
+        .and(path("/"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(serde_json::json!({
+            "version": 1,
+            "stage": "SupergraphRequest",
+            "control": "continue",
+            "context": {
+                "entries": {
+                    "apollo::authorization::required_policies": {
+                        // NOTE: see the note above, but this shouldn't govern how the test
+                        // behaves; it's the context object that does the dirt for the TestHarness.
+                        // Change this value if you use it with the IntegrationTest builder
+                        "admin": false
+                    }
+                }
+            }
+        })))
+        .mount(&mock_coprocessor)
+        .await;
+
+    let mut subgraphs = MockedSubgraphs::default();
+    subgraphs.insert(
+        "subgraph_a",
+        MockSubgraph::builder()
+            .with_json(
+                serde_json::json!({"query": "{private{id}}"}),
+                serde_json::json!({"data": {"private": {"id": "123"}}}),
+            )
+            .with_json(
+                serde_json::json!({"query": "{public{id}}"}),
+                serde_json::json!({"data": {"public": {"id": "456"}}}),
+            )
+            .with_json(
+                serde_json::json!({"query": "{opensecret{id}}"}),
+                serde_json::json!({"data": {"opensecret": {"id": "789"}}}),
+            )
+            .build(),
+    );
+
+    let supergraph_harness = TestHarness::builder()
+        .configuration_json(serde_json::json!({
+            "coprocessor": {
+                "url": coprocessor_address,
+                "supergraph": {
+                    "request": {
+                        "context": "all"
+                    }
+                }
+            },
+            "include_subgraph_errors": {
+                "all": true
+            }
+        }))
+        .unwrap()
+        .schema(include_str!(
+            "../../fixtures/directives/policy/policy_schema_with_interfaces.graphql"
+        ))
+        .extra_plugin(subgraphs)
+        .build_supergraph()
+        .await
+        .unwrap();
+
+    let context = Context::new();
+    context
+        // NOTE: there is no `admin` policy in the context
+        .insert("apollo::authorization::required_policies", json! { [] })
+        .unwrap();
+
+    // WHEN
+    //   * we make a request to an implementor without the policy directive
+    let request = supergraph::Request::fake_builder()
+        .query(r#"{ opensecret { id } }"#)
+        .context(context)
+        .method(Method::POST)
+        .build()
+        .unwrap();
+
+    // THEN
+    //   * we... get the data? expected? not sure
```

Comment:
are we still unsure? or has Derek's analysis on the issue cleared it up?

Reply:
oh, cool, I didn't know about `should_panic`; added! and, added a todo mentioning fed-790; will comment on the ticket, too